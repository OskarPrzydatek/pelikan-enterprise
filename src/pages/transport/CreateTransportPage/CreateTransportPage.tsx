import { SubmitHandler, useForm } from 'react-hook-form';
import { useNavigate } from 'react-router-dom';

import { fetchPost } from '~/api';
import { CreateTransportTemplate } from '~/components/templates';
import { Endpoints, Slugs, transportTypeOptions } from '~/constants';
import { ITransport } from '~/models';
import { errorNotification, successNotification } from '~/notifications';

export const CreateTransportPage: React.FC = () => {
  const methods = useForm<ITransport>();
  const navigate = useNavigate();

  const onSubmit: SubmitHandler<ITransport> = async (data) => {
    const response = await fetchPost<ITransport>(
      Endpoints.CREATE_TRANSPORT,
      data
    );

    if (response.ok) {
      successNotification('Transport dodano pomy≈õlnie');
      navigate(`/${Slugs.TRANSPORT_OVERVIEW}`);
      return;
    }

    errorNotification();
  };

  return (
    <CreateTransportTemplate
      createTransportFormMethods={methods}
      transportTypeOptions={transportTypeOptions}
      onSubmitCreateTransport={methods.handleSubmit(onSubmit)}
    />
  );
};
