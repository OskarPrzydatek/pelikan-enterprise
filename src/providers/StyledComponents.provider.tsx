import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from '../styles/layouts';
import { theme } from '../styles/theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};
