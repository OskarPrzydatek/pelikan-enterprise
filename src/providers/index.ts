export { StyledComponentsProvider } from './StyledComponents.provider';
