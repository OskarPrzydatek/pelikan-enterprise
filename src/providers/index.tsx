import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from '../styles/layouts';
import { theme } from '../styles/theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

interface ITestingPageProvider {
  element: JSX.Element;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};

export const TestingPageProvider: React.FC<ITestingPageProvider> = ({
  element,
}: ITestingPageProvider) => {
  return (
    <StyledComponentsProvider>
      <MemoryRouter initialEntries={['/page1']}>
        <Routes>
          <Route element={element} path="/:page" />
        </Routes>
      </MemoryRouter>
    </StyledComponentsProvider>
  );
};
