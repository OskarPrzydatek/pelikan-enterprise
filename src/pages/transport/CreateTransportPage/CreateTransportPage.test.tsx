import { render } from '@testing-library/react';
import { describe } from 'vitest';

import { TestingPageProvider } from '~/providers';

import { CreateTransportPage } from './CreateTransportPage';

const MockCreateTransportPage: React.FC = () => (
  <TestingPageProvider element={<CreateTransportPage />} />
);

describe('CreateTransportPage', () => {
  test('page snapshot', () => {
    const view = render(<MockCreateTransportPage />);
    expect(view).toMatchSnapshot();
  });
});
