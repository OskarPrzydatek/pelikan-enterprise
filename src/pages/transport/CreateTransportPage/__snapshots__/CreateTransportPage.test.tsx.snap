// Vitest Snapshot v1

exports[`CreateTransportPage > page snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gswNZR cJrdyV"
      >
        <form
          class="sc-jIRcFI iwFYYu"
        >
          <h2
            class="sc-eDvSVe dcXtK"
          >
            Dodaj transport
          </h2>
          <div
            class="sc-dmctIk iThEeJ"
          >
            <div
              class="sc-kgTSHT fmvgAY"
            >
              <input
                class="sc-fLcnxK gGYQKp"
                data-testid="create-company-name"
                id="name"
                name="name"
                placeholder="Nazwa firmy transportowej"
              />
            </div>
          </div>
          <div
            class="sc-iveFHk hxGauG"
          >
            <div
              class="sc-cwSeag fbfUTf"
            >
              <span
                class="sc-iJnaPW ixWkhk"
              >
                <span
                  class="sc-gKPRtg dQATfs"
                >
                  <svg
                    data-testid="chevron-icon"
                    fill="none"
                    viewBox="0 0 27 16"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M2 2L13.5 13L25 2"
                      stroke="#013059"
                      stroke-width="3"
                    />
                  </svg>
                </span>
              </span>
              <select
                class="sc-gikAfH koRAmy"
                data-testid="create-transport-transport-kind"
                name="transportType"
              >
                <option
                  class="sc-ezOQGI"
                  disabled=""
                  selected=""
                  value=""
                >
                  Rodzaj transportu
                </option>
                <option
                  class="sc-ezOQGI"
                  value="PLANE"
                >
                  Samolot
                </option>
                <option
                  class="sc-ezOQGI"
                  value="BUS"
                >
                  Autobus
                </option>
                <option
                  class="sc-ezOQGI"
                  value="SHIP"
                >
                  Statek
                </option>
              </select>
            </div>
          </div>
          <div
            class="sc-bjfHbI kTUcsA"
          >
            <div
              class="sc-idXgbr enkruV"
            >
              <textarea
                class="sc-hHTYSt lavPOm"
                data-testid="create-transport-desription"
                id="description"
                name="description"
                placeholder="Opis"
              />
            </div>
          </div>
          <input
            class="sc-hhOBVt dvAfkB"
            type="submit"
            value="Dodaj"
          />
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gswNZR cJrdyV"
    >
      <form
        class="sc-jIRcFI iwFYYu"
      >
        <h2
          class="sc-eDvSVe dcXtK"
        >
          Dodaj transport
        </h2>
        <div
          class="sc-dmctIk iThEeJ"
        >
          <div
            class="sc-kgTSHT fmvgAY"
          >
            <input
              class="sc-fLcnxK gGYQKp"
              data-testid="create-company-name"
              id="name"
              name="name"
              placeholder="Nazwa firmy transportowej"
            />
          </div>
        </div>
        <div
          class="sc-iveFHk hxGauG"
        >
          <div
            class="sc-cwSeag fbfUTf"
          >
            <span
              class="sc-iJnaPW ixWkhk"
            >
              <span
                class="sc-gKPRtg dQATfs"
              >
                <svg
                  data-testid="chevron-icon"
                  fill="none"
                  viewBox="0 0 27 16"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M2 2L13.5 13L25 2"
                    stroke="#013059"
                    stroke-width="3"
                  />
                </svg>
              </span>
            </span>
            <select
              class="sc-gikAfH koRAmy"
              data-testid="create-transport-transport-kind"
              name="transportType"
            >
              <option
                class="sc-ezOQGI"
                disabled=""
                selected=""
                value=""
              >
                Rodzaj transportu
              </option>
              <option
                class="sc-ezOQGI"
                value="PLANE"
              >
                Samolot
              </option>
              <option
                class="sc-ezOQGI"
                value="BUS"
              >
                Autobus
              </option>
              <option
                class="sc-ezOQGI"
                value="SHIP"
              >
                Statek
              </option>
            </select>
          </div>
        </div>
        <div
          class="sc-bjfHbI kTUcsA"
        >
          <div
            class="sc-idXgbr enkruV"
          >
            <textarea
              class="sc-hHTYSt lavPOm"
              data-testid="create-transport-desription"
              id="description"
              name="description"
              placeholder="Opis"
            />
          </div>
        </div>
        <input
          class="sc-hhOBVt dvAfkB"
          type="submit"
          value="Dodaj"
        />
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
